main/views.py

```diff
@@ -36,38 +43,101 @@ def search():
+class CompanyProfile(TemplateView):
+    template_name = 'company/Profile.html'
 
+    def __init__(self):
+        super(CompanyProfile, self).__init__()
+        self.org = None
+        self.specialist = None
 
-@app.route('/service_activity/confirm/<token>')
-def confirm_specialist_activity(token):
-    activity_id = confirm_token(token)
-    activity = ServiceActivity.query.filter_by(id=activity_id).first_or_404()
-    if activity.confirmed:
-        flash('Activity already confirmed.')
-    else:
-        activity.confirmed = True
-        flash('You have confirmed your relationship with {}.'.format(
-            activity.specialist.name))
+    def get(self, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+    def get_context_data(self, **kwargs):
+        context = super(CompanyProfile, self).get_context_data()
+        context.update({'company': self.get_company(kwargs)})
+        # context.update({'specialist': self.get_specialist(kwargs)})
+
+        return context
+
+    def get_company(self, kwargs):
+        self.org = Company.query.get(kwargs.get('company_id'))
+        return self.org
+
+
+app.add_url_rule(
+    '/company/<int:company_id>/profile',
+    view_func=CompanyProfile.as_view('company_profile')
+)
+
+
+class UserProfile(TemplateView):
+    template_name = 'user/Profile.html'
+
+    def __init__(self):
+        super(UserProfile, self).__init__()
+        self.user = None
+
+    def get(self, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+    def get_context_data(self, **kwargs):
+        context = super(UserProfile, self).get_context_data()
+        context.update({'user': self.get_user(kwargs)})
+        context.update({'current_user': current_user})
+        context.update({'activity': self.get_activity(kwargs)})
+        context.update({'session': session})
+        context.update({'form': self.get_service_activity_form() })
+
+        return context
 
-    return render_template('ConfirmServiceActivity.html')
+    def get_service_activity_form(self):
+        form = None
+        if self.user.specialist:
+            form = AddServiceActivityForm.get_form(self.user.specialist)
 
+        return form
 
-@app.route('/service_activity/add', methods=['GET', 'POST'])
-def add_service_activity():
-    form = AddServiceActivityForm()
+    def get_user(self, kwargs):
+        self.user = User.query.get(kwargs.get('user_id'))
+        return self.user
 
-    if request.method == 'POST' and form.validate():
-        specialist = Specialist.query.get(form.specialist_id.data)
-        customer = Customer.query.get(form.customer_id.data)
-        service = Service.query.get(form.service_id.data)
+    def get_activity(self, kwargs):
+        if self.user is not None:
+            if self.user.specialist is not None:
+                self.activity = UserUserActivity.query.filter_by(
```

Comment:
I prefer to start `filter_by` on the next line, it looks clearer, but it's a question of style:
````
self.activity = UserUserActivity.query \
    .filter_by(from_user_id=....```